extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextProcessor.java

```diff
@@ -102,6 +106,28 @@ CamelContextBuildItem context(
         return new CamelContextBuildItem(context);
     }
 
+    /**
+     * This step customizes camel context for development mode.
+     *
+     * @param recorder     the recorder
+     * @param capabilities the registered capabilities
+     * @param producer     producer of context customizer build item
+     */
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public void developmenMode(
+            CamelContextRecorder recorder,
+            Capabilities capabilities,
+            BuildProducer<CamelContextCustomizerBuildItem> producer) {
+        if (capabilities.isPresent(CamelCapabilities.MAIN)) {
+            String val = CamelSupport.getOptionalConfigValue("camel.main.shutdownTimeout", String.class, null);
+            if (val == null) {
+                //if no graceful timeout is set in development mode, graceful shutdown is replaced with no shutdown
+                producer.produce(new CamelContextCustomizerBuildItem(recorder.createNoShutdownStrategyCustomizer()));
+            }
+        }
```

Comment:
Shouldn't we also set no shutdown strategy in DEV mode when no running with main ?

Reply:
makes sense, I'll fix it.

Reply:
fixed